web/client/utils/MapInfoUtils.js

```diff
@@ -21,21 +21,95 @@ import vector from './mapinfo/vector';
 import threeDTiles from './mapinfo/threeDTiles';
 
 let MapInfoUtils;
+/**
+ * Map of info modes which are used to display feature info data (identify tools).
+ * To be distinguished with INFO_FORMATS which is the map of mime types used in client server communication.
+ * These are strictly a representation of the various ways that info data is visualized.
+ * Has an N <=> N relationship with INFO_FORMATS.
+ */
+const INFO_VIEW_MODES = {
+    TEXT: "TEXT",
+    PROPERTIES: "PROPERTIES",
+    HTML: "HTML",
+    TEMPLATE: "TEMPLATE"
+};
+
+/**
+ * @returns {object} Map of views which are used to display feature info data (identify tools).
+ */
+export const getInfoViewModes = () => {
+    return {...INFO_VIEW_MODES};
+};
+/**
+ * @param {string} infoFormat the info format mime type.
+ * @returns {string} the info view mode that is used for that info format.
+ */
+export const getInfoViewByInfoFormat = (infoFormat) => {
+    let infoView;
+    switch (infoFormat) {
+    case INFO_FORMATS.TEXT:
+        infoView = INFO_VIEW_MODES.TEXT;
+        break;
+    case INFO_FORMATS.HTML:
+        infoView = INFO_VIEW_MODES.HTML;
+        break;
+    case INFO_FORMATS.JSON:
+        infoView = INFO_VIEW_MODES.PROPERTIES;
+        break;
+    case INFO_FORMATS.GEOJSON:
+        infoView = INFO_VIEW_MODES.PROPERTIES;
+        break;
+    default:
+        // TODO: re-assess leaving default null value, this way tests work but caller is burdened with fallback.
+        infoView;
+    }
+
+    return infoView;
+};
+/**
+ * @param {string} infoView the info view mode.
+ * @param {array} layerInfoFormatCfg the layer supported GFI mime types.
+ * @returns {string} the info format mime type.
+ */
+export const getInfoFormatByInfoView = (infoView, layerInfoFormatCfg) => {
+    let infoFormat;
+    let isGeoJSON = layerInfoFormatCfg?.includes(GEOJSON_MIME_TYPE);
+    switch (infoView) {
+    case INFO_VIEW_MODES.TEXT:
+        infoFormat = INFO_FORMATS.TEXT;
+        break;
+    case INFO_VIEW_MODES.HTML:
+        infoFormat = INFO_FORMATS.HTML;
+        break;
+    case INFO_VIEW_MODES.PROPERTIES:
+        infoFormat = isGeoJSON ? INFO_FORMATS.GEOJSON : INFO_FORMATS.JSON;
+        break;
+    case INFO_VIEW_MODES.TEMPLATE:
+        infoFormat = isGeoJSON ? INFO_FORMATS.GEOJSON : INFO_FORMATS.JSON;
```

Comment:
isGeoJSON makes sense only in this block.
```suggestion
        let isGeoJSON = layerInfoFormatCfg?.includes(GEOJSON_MIME_TYPE);
        infoFormat = isGeoJSON ? INFO_FORMATS.GEOJSON : INFO_FORMATS.JSON;
```

Reply:
Fixed in https://github.com/geosolutions-it/MapStore2/pull/9917/commits/9b24f0df5edc2fe5a789f504b9c283475838a901